tap_spotify/streams.py

```diff
@@ -162,3 +162,14 @@ class GlobalViralTracksDailyStream(_PlaylistTracksStream):
     name = "global_viral_tracks_daily_stream"
     path = "/playlists/37i9dQZEVXbLiRSasKsNU9"
     primary_keys = ["rank", "synced_at"]
+
+
+class UserSavedTracksStream(SpotifyStream):
+    """Define global viral tracks daily stream."""
+
+    name = "user_saved_track_streams"
+    path = "/me/tracks"
+    primary_keys = ["synced_at"]
```

Comment:
```suggestion
    primary_keys = ["id", "synced_at"]
```

The value of `synced_at` is established once at runtime and set across all records in the stream, which causes a constrain violation error for duplicate primary keys when loading to Postgres (for example). We can form a composite key using `id` and `synced_at` to resolve this.